spec/features/users/webauthn_management_spec.rb

```diff
@@ -109,25 +109,42 @@
 
       expect(page).to have_content 'key1'
 
-      click_button t('account.index.webauthn_delete')
+      click_link t('account.index.webauthn_delete')
+
+      expect(current_path).to eq webauthn_setup_delete_path
+
+      click_button t('account.index.webauthn_confirm_delete')
 
       expect(page).to_not have_content 'key1'
       expect(page).to have_content t('notices.webauthn_deleted')
     end
 
-    it 'prevents a user from deleting the last key' do
+    it 'allows the user to cancel delete the webauthn key' do
       create_webauthn_configuration(user, 'key1', '1', 'foo1')
 
       sign_in_and_2fa_user(user)
-      PhoneConfiguration.first.update(mfa_enabled: false)
       visit account_path
 
       expect(page).to have_content 'key1'
 
-      click_button t('account.index.webauthn_delete')
+      click_link t('account.index.webauthn_delete')
+
+      expect(current_path).to eq webauthn_setup_delete_path
+
+      click_link t('links.cancel')
+
+      expect(page).to have_content 'key1'
+    end
+
+    it 'prevents a user from deleting the last key' do
```

Comment:
Would it be clearer to specify that you can't delete the last key if it's your last remaining 2FA option? Should we also have a test that makes sure you can delete all your security keys if you have other 2FA options configured?

Reply:
How precisely would we do this: "Would it be clearer to specify that you can't delete the last key if it's your last remaining 2FA option?" ?  We used to allow you to delete it and then post a message that you can't.

Reply:
Sorry for the confusion. For the first part, I meant to update the text of the test title to make it clearer what scenario is being tested. You're right about the second part. I missed the user setup at the top that also gives them a phone as a 2FA method.